database/dialect.go

```diff
@@ -0,0 +1,24 @@
+package database
+
+import (
+	"strings"
+
+	"github.com/cloudquery/cq-provider-sdk/helpers"
+	"github.com/cloudquery/cq-provider-sdk/provider/schema"
+)
+
+func DSNtoDialect(dsn string) (d schema.DialectType, newDSN string, err error) {
```

Comment:
ParseDialectDSN seems a better name, also why isn't the helper function just here? feels misplaced there.

Reply:
Renamed. Helper function `ParseConnectionString` is used in multiple places, and moving this into `helpers` isn't possible as `schema` is needed for `DialectType` reference. Could move all (helper + this) into `schema` but that would be a step back compared to the current situation.